domainservice/src/main/kotlin/systems/ajax/malov/stockanalyzer/mapper/proto/StockPriceMapper.kt

```diff
@@ -0,0 +1,28 @@
+package systems.ajax.malov.stockanalyzer.mapper.proto
+
+import com.google.protobuf.Timestamp
+import com.squareup.wire.Instant
+import systems.ajax.malov.internalapi.output.pubsub.stock.stock_price.proto.StockPrice
+import systems.ajax.malov.stockanalyzer.entity.MongoStockRecord
+import systems.ajax.malov.stockanalyzer.mapper.proto.BigDecimalProtoMapper.convertToBigDecimalProto
+
+object StockPriceMapper {
+    fun MongoStockRecord.toStockPrice(): StockPrice {
+        return StockPrice.newBuilder()
+            .setPrice(convertToBigDecimalProto(currentPrice))
+            .setStockSymbolName(symbol)
+            .setTimestamp(
+                dateOfRetrieval?.let {
+                    toTimestampProto(it)
+                } ?: Timestamp.getDefaultInstance()
+            )
+            .build()
+    }
+
+    private fun toTimestampProto(dateOfRetrieval: Instant): Timestamp {
+        return Timestamp.newBuilder()
+            .setSeconds(dateOfRetrieval.epochSecond)
+            .setNanos(dateOfRetrieval.nano)
+            .build()
+    }
```

Comment:
```suggestion
        return StockPrice.newBuilder()
            .setPrice(convertToBigDecimalProto(currentPrice))
            .setStockSymbolName(symbol)
            .apply { dateOfRetrieval?.toTimestampProto()?.let { setTimestamp(it) } }
            .build()
    }

    private fun Instant.toTimestampProto(): Timestamp {
        return Timestamp.newBuilder()
            .setSeconds(epochSecond)
            .setNanos(nano)
            .build()
    }
```
By the way, TimestampProto can be extracted from a separate mapper. in future, it may be reused as this is common mapping